ts/spi/provider_frame.ts

```diff
@@ -221,7 +229,7 @@ export class ProviderFrame {
       options: CredentialHintOptions) {
     console.info('Handling hint request');
 
-    let hints = await this.getHints(options);
+    let hints = await this.race([], this.getHints(options));
```

Comment:
I am not sure I understand the concern. When you call `await`, subsequent interactions with the result of the await are being passed on to you via a generator's `next(...)` method. So adding it to the top level promise will cascade to all future promises underneath the `await` call. 

Are you talking about something else ?